Tests/ButterTests/Mocks/MockURLSession.swift

```diff
@@ -3,7 +3,7 @@
 //
 
 import Foundation
-import Butter
+@testable import Butter
 
 class MockURLSession_Success: URLSessionDataTaskInterface {
 	func dataTask(with request: URLRequest, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> URLSessionDataTask {
```

Comment:
alright just needed to read a bit further down. so you're basically just mocking part of the behavior of the URLSession. 

have you thought about using `URLProtocol` to stub network responses instead of mocking session behavior? 👀

Reply:
I'll look into this. Initially I figured since Butter was interacting directly with session that would be a good starting point but typically i have seen people use stubs with URLProtocol